### DISCUSSION TITLE
[Fantom] Can't sign Fantom transaction in Swift

### DISCUSSION THREAD

**Discussion by eaglean** [trustwallet/wallet-core#1795] [Q&A]
Hello, TrustWallet dev team!
I can't sign a transaction with Fantom network.

### Detail:

I'm going to transfer FTMO tokens by using `WalletCore` iOS SDK.
I signed a Fantom transaction but getting error when I submitted the hex string of the signed transaction on the online FTM broadcast site - https://ftmscan.com/pushTx.

```
guard let myWallet = HDWallet(mnemonic: "xxxx xxxx xxxx xxxx xxxx xxxx xxxx xxxx xxxx", passphrase: "") else { return }
let input = EthereumSigningInput.with {
  $0.chainID = Data(hexString: "FA")! // 250
  $0.nonce = Data(hexString: "00")!
  $0.gasLimit = Data(hexString: "0135B7")! // 79287
  $0.toAddress = "0x9bd0611610a0f5133e4dd1bfdd71c5479ee77f37" // FTMO
  $0.privateKey = myWallet.getKeyForCoin(coin: .fantom).data
  $0.transaction = EthereumTransaction.with {
    $0.erc20Transfer = EthereumTransaction.ERC20Transfer.with {
      $0.to = "0xZZZZZZZZZZZZZZZZZZZZZZZZZZZ" // Destination wallet address
      $0.amount = Data(hexString: "1bc16d674ec80000")! // 2000000000000000000
    }
  }
}

let output: EthereumSigningOutput = AnySigner.sign(input: input, coin: .fantom)
let hexString = output.encoded.hexString
```

Error message on https://ftmscan.com/pushTx
`Error! Unable To Broadcast Txn : {"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"transaction underpriced"}}`

### Question:
- What is wrong in the above code?
- Can you please share the test code working with Fantom networks? (I just saw you are missing just that - https://github.com/trustwallet/wallet-core/tree/master/swift/Tests/Blockchains)

Thanks,
Jiang

**Comment by optout21** [comment]
"transaction underpriced" usually indicated that gasPrice is too low.  Oops, you didn't set gasPrice!?

**Reply by eaglean** [reply]
@catenocrypt I didn't set `gasPrice`, what should be that?

**Reply by eaglean** [reply]
@catenocrypt I did set the `gasPrice` but same issue.
`$0.gasPrice = Data(hexString: "09c7652400")! // 42000000000`

`Error! Unable To Broadcast Txn : {"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"transaction underpriced"}}`

**Reply by hewigovens** [reply]
take a look at https://ftmscan.com/gastracker, make sure your gas price is above average
